src/modules/scraper/adapter/messaging/FillEventsConsumer.ts

```diff
@@ -45,7 +45,14 @@ export class FillEventsConsumer {
       throw new Error("Deposit not found in db");
     }
 
-    if (deposit.fillTxs.includes({ hash: transactionHash, totalFilledAmount })) return;
+    const fillTxIndex = deposit.fillTxs.findIndex(
+      (fillTx) => fillTx.hash === transactionHash && fillTx.totalFilledAmount === totalFilledAmount,
```

Comment:
Isn't transactionHash alone sufficient to identify uniquely the fill?

Reply:
I remember you said that we can have multiple fills in a tx

Reply:
Ah you're right. So couldn't you then just check totalFilledAmlunt?

Reply:
yes, I can. I added the `transactionHash` as being part of the check because I need it anyway to be stored in db

Reply:
hm ok